cpp/src/slice2java/Gen.cpp

```diff
@@ -1960,22 +1982,34 @@ Slice::Gen::TypesVisitor::visitClassDefEnd(const ClassDefPtr& p)
     out << eb;
 
     // Implementation of 'ice_staticId'.
-    out << sp << nl << "public static String ice_staticId()";
+    out << sp;
+    out << nl << "/**";
+    out << nl << " * Gets the type ID of the associated Slice interface";
+    out << nl << " *";
+    out << nl << " * @return the string \"" << p->scoped() << "\"";
+    out << nl << " */";
+    out << nl << "public static String ice_staticId()";
     out << sb;
     out << nl << "return \"" << p->scoped() << "\";";
     out << eb;
 
     // Implementation of 'ice_id'.
-    out << sp << nl << "@Override";
+    out << sp;
+    writeDocComment(out, "{@inheritDoc}");
+    out << nl << "@Override";
     out << nl << "public String ice_id()";
     out << sb;
     out << nl << "return ice_staticId();";
     out << eb;
 
     // Implementation of '_iceWriteImpl'.
     int iter = 0;
+
     out << sp;
-    writeHiddenDocComment(out);
+    writeDocComment(
+        out,
+        "{@inheritDoc}\n"
```

Comment:
We need `@inheritdoc` + `@hidden` because just `@hidden` overrides the parent's doc-comment?